spec/bolt/module_installer_spec.rb

```diff
@@ -54,6 +54,17 @@
       )
     end
 
+    it 'errors if module is already in puppetfile without version' do
+      File.write(puppetfile, new_module)
+      expect {
+        installer.add(new_module, specs, puppetfile, moduledir, project_file, install_config)
```

Comment:
```suggestion
        installer.add(new_module, [], puppetfile, moduledir, project_file, install_config)
```

We should pass an empty array for specs here. The specs are the modules listed in project config (see [this doc](https://puppet.com/docs/bolt/latest/bolt_installing_modules.html#manually-specify-modules-in-a-bolt-project)). By passing an empty array for the specs, we won't hit the first error that ensures all specs are present in the Puppetfile.